contracts/interchain-token-service/InterchainTokenService.sol

```
@@ -335,74 +270,45 @@ contract InterchainTokenService is
      * @param salt the salt to be used.
      * @param name the name of the token to be deployed.
      * @param symbol the symbol of the token to be deployed.
      * @param decimals the decimals of the token to be deployed.
      * @param distributor the address that will be able to mint and burn the deployed token.
-     * @param mintTo The address where the minted tokens will be sent upon deployment
-     * @param mintAmount The amount of tokens to be minted upon deployment
-     * @param operator_ The operator data for standardized tokens
      * @param destinationChain the name of the destination chain to deploy to.
      * @param gasValue the amount of native tokens to be used to pay for gas for the remote deployment. At least the amount
      * specified needs to be passed to the call
      * @dev `gasValue` exists because this function can be part of a multicall involving multiple functions that could make remote contract calls.
      */
     function deployInterchainToken(
         bytes32 salt,
+        string calldata destinationChain,
         string memory name,
         string memory symbol,
         uint8 decimals,
         bytes memory distributor,
-        bytes memory mintTo,
-        uint256 mintAmount,
-        bytes memory operator_,
-        string calldata destinationChain,
+        bytes memory operator,
         uint256 gasValue
     ) external payable whenNotPaused {
         bytes32 tokenId_ = tokenId(msg.sender, salt);
-        _deployRemoteStandardizedToken(
-            tokenId_,
-            name,
-            symbol,
-            decimals,
-            distributor,
-            mintTo,
-            mintAmount,
-            operator_,
-            destinationChain,
-            gasValue
-        );
+
+        if (bytes(destinationChain).length == 0) {
+            address tokenAddress_;
+            tokenAddress_ = _deployInterchainToken(tokenId_, distributor, name, symbol, decimals);
```

Should we fail early if token manager was already deployed for this salt?

---

we could but what is the point, really?